lua/ulx/modules/sh/cfc_brazil.lua

```diff
@@ -0,0 +1,141 @@
+        height = 14500 / 2,
+        fallback = Vector( 9192, 10270, -10897 ),
+        offsetX = 900,
+        offsetY = 1900
+    },
+    ["gm_bigcity_improved"] = {
+        centerZ = 14500 / 2 - 11400,
+        height = 14500 / 2,
+        fallback = Vector( 9192, 10270, -10897 ),
+        offsetX = 900,
+        offsetY = 1900
+    },
+    ["gm_bigcity_improved_lite"] = {
+        centerZ = 14500 / 2 - 11400,
+        height = 14500 / 2,
+        fallback = Vector( 9192, 10270, -10897 ),
+        offsetX = 900,
+        offsetY = 1900
+    },
+    ["gm_construct"] = {
+        centerZ = 10500 / 2 - 500,
+        height = 10500 / 2,
+        fallback = Vector( -2069, -3316, -252 ),
+        offsetX = 1000,
+        offsetY = 500
+    },
+    ["gm_novenka"] = {
+        centerZ = 12800 / 2 - 2050,
+        height = 12800 / 2,
+        fallback = Vector( 9760, -13800, -700 ),
+        offsetX = 800,
+        offsetY = 500
+    },
+    ["gm_bluehills_test3"] = {
+        centerZ = 14400 / 2 - 1100,
+        height = 14400 / 2,
+        fallback = Vector( -8750, -8500, -1084 ),
+        offsetX = 500,
+        offsetY = 500
+    }, 
+    ["gm_flatgrass"] = {
+        centerZ = 28200 / 2 - 12800,
+        height = 28200 / 2,
+        fallback = Vector( -700, 100, -12764 ),
+        offsetX = 250,
+        offsetY = 500
+    }, 
+    ["gm_functional_flatgrass"] = {
+        centerZ = 28600 / 2 - 16300,
+        height = 28600 / 2,
+        fallback = Vector( 10500, -9400, -16252 ),
+        offsetX = 3500,
+        offsetY = 3500
+    }, 
+    ["gm_genesis"] = {
+        centerZ = 27600 / 2 - 15300,
+        height = 27600 / 2,
+        fallback = Vector( 6800, 6500, -14974 ),
+        offsetX = 600,
+        offsetY = 600
+    }
+}
+
+local function planTrip( ply )
+    for _ = 1, 20 do
+        local mapData = MAP_POSITION_DATA[game.GetMap()]
+        local pos
+        
+        if mapData then
+            pos = Vector( math.random( MIN_X, MAX_X ), math.random( MIN_Y, MAX_Y ), mapData.centerZ + math.random( -mapData.height, mapData.height ) )
+        else
+            pos = Vector( math.random( MIN_X, MAX_X ), math.random( MIN_Y, MAX_Y ), math.random( MIN_Z, MAX_Z ) )
+        end
+        
+        if util.IsInWorld( pos ) then
+            local minHullSize, maxHullSize = ply:GetCollisionBounds()
+            local minHull = Vector( minHullSize.x - HULL_BUFFER, minHullSize.y - HULL_BUFFER, 0 )
+            local maxHull = Vector( maxHullSize.x + HULL_BUFFER, maxHullSize.y + HULL_BUFFER, 0.001 )
+            
+            local validationTrace = util.TraceHull( {
+                start = pos,
+                endpos = pos + Vector( 0, 0, maxHullSize.z ),
+                mins = minHull,
+                maxs = maxHull
+            } )
+            
+            if not validationTrace["Hit"] then
+                local floorTrace = util.TraceHull( {
+                    start = pos,
+                    endpos = pos + Vector( 0, 0, MIN_Z ),
```

Comment:
you might not want to use MIN_Z here
this would only really work out if MIN_Z is negative. it is in the case but if I understand correctly MIN_Z is a config value representing the minimum Z value that you get a random position from so it doesn't necessarily have to be negative

Reply:
MIN_Z isn't a config variable, it and the other min/max constants are from the max map size for gmod.

Reply:
yeah, I wasn't aware that is what those numbers where :+1: